courses/migrations/0029_revert_certificates_prior_aug_8.py

```diff
@@ -0,0 +1,129 @@
+# Generated by Django 3.2.15 on 2022-10-26 15:40
+
+from django.db import migrations
+from datetime import datetime, timezone
+from cms.models import CertificatePage
+from courses.models import CourseRunCertificate, ProgramCertificate
+
+
+AUGUST_8_2022 = datetime(2022, 8, 8, tzinfo=timezone.utc)
+SEPTEMBER_20_2022 = datetime(2022, 9, 20, tzinfo=timezone.utc)
+
+
+def get_course_certificate_cms_page(course_run_cert):
+    """Try to get the CMS page for a course certificate"""
+    return (
+        course_run_cert.course_run.course.page.get_children()
+        .type(CertificatePage)
+        .live()
+        .first()
+        if course_run_cert.course_run.course.page
+        else None
+    )
+
+
+def get_program_certificate_cms_page(program_cert):
+    """Try to get the CMS page for a program certificate"""
+    return (
+        program_cert.program.page.certificate_page
+        if program_cert.program.page
+        else None
+    )
+
+
+def get_latest_revision_by_date(certificate_page, max_date):
+    """Get the latest certificate page revision created till max_date"""
+    return (
+        certificate_page.revisions.filter(created_at__lte=max_date)
+        .order_by("-created_at")
+        .first()
+    )
```

Comment:
All these utils seem to be useful in the future as well and the same kind of code is written in the `save` method for CourseRunCertificate and ProgramCertificate models. Following DRY, How about moving these to a common place and using them here and in save?

Reply:
Good point and a good catch, I had a similar thing in mind when I converted these into functions and while looking at the scope, and since this is purely just a data migration so I was thinking of keeping it atomic in a way that the testing scope of this PR and the ability to revert in case of unexpected results might be simple and straight forward. So the idea was to plan a refactor out of this PR's scope and move these functions into utils.

But i guess, Maybe, maybe we should do this here 🤔

Reply:
As we discussed in a call, Let's keep these here for now in this PR. We will be creating a follow-up PR for the refactoring as that will increase the testing scope and will be hard to test. This PR looks good to me. We can merge this one for now.